app/models/managed_interactive.rb

```diff
@@ -0,0 +1,269 @@
+      inherit_full_window: inherit_full_window,
+      custom_full_window: custom_full_window,
+      inherit_click_to_play_prompt: inherit_click_to_play_prompt,
+      custom_click_to_play_prompt: custom_click_to_play_prompt,
+      inherit_image_url: inherit_image_url,
+      custom_image_url: custom_image_url
+    }
+  end
+
+  def portal_hash
+    iframe_data = to_hash
+    iframe_data[:type] = 'iframe_interactive'
+    iframe_data[:id] = id
+    iframe_data[:display_in_iframe] = reportable_in_iframe?
+    iframe_data
+  end
+
+  def report_service_hash
+    {
+      type: 'iframe_interactive',
+      id: embeddable_id,
+      name: name,
+      url: url,
+      show_in_featured_question_report: show_in_featured_question_report,
+      display_in_iframe: reportable_in_iframe?,
+      width: native_width,
+      height: native_height,
+      question_number: index_in_activity
+    }
+  end
+
+  def duplicate
+    # Generate a new object with those values
+    ManagedInteractive.new(self.to_hash)
+    # N.B. the duplicate hasn't been saved yet
+  end
+
+  def export
+    json = self.as_json(only:[:name,
+      :url_fragment,
+      :authored_state,
+      :is_hidden,
+      :is_full_width,
+      :show_in_featured_question_report,
+      :inherit_aspect_ratio_method,
+      :custom_aspect_ratio_method,
+      :inherit_native_width,
+      :custom_native_width,
+      :inherit_native_height,
+      :custom_native_height,
+      :inherit_click_to_play,
+      :custom_click_to_play,
+      :inherit_full_window,
+      :custom_full_window,
+      :inherit_click_to_play_prompt,
+      :custom_click_to_play_prompt,
+      :inherit_image_url,
+      :custom_image_url
+    ])
+    if library_interactive
+      json["library_interactive"] = {
+        hash: library_interactive.generate_export_hash(),
+        data: library_interactive.export()
+      }.to_json()
+    end
+  end
+
+  def self.import(import_hash)
+    # save off the imported library interactive to hydrate it after the instance is created
+    imported_library_interactive = import_hash[:library_interactive]
+    import_hash.delete(:library_interactive)
+    managed_interactive = self.new(import_hash)
+
+    # find the existing matching library interactive or create a new one
+    if imported_library_interactive
+      library_interactive = LibraryInteractive.find_by_export_hash(imported_library_interactive[:hash])
+      if library_interactive
+        self.library_interactive = library_interactive
+      else
+        self.library_interactive = LibraryInteractive.import(imported_library_interactive[:data])
+
+        # REVIEW QUESTION: should we call #save here - if we don't and the user imports an activity
+        # that uses the same library interactive more than once it won't find the newly created
+        # library interactive
+      end
+    end
+
+    managed_interactive
+  end
+
+  # REVIEW TODO: is update_labbook_options needed?
```

Comment:
Lets take this out but put a comment in the MwInteractive code indicating this is not supported by the new ManagedInteractive. So we need to migrate this feature to a better solution.

I'll add a story about this in the hopes we can get to it before putting this down.

Reply:
Done.